source/connections-survive-step-down/tests/README.rst

```diff
@@ -0,0 +1,214 @@
+  a primary.
+- Using the same connection that was used for the initial find, continue
+  iterating the collection by issuing a getMore operation.
+- Verify that the next batch of results is returned successfully.
+
+Drivers that do not ordinarily allow applications to execute operations
+on specific connections may need to perform the getMore iteration test on
+a lower level than the public API exposed to applications.
+In this case the driver SHOULD verify that both initial find and the
+subsequent getMore are dispatched on the same connection, for example by
+comparing socket handles used for the two operations.
+
+Acknowledged Write
+``````````````````
+
+This test requires server version 4.2 or higher.
+
+Perform the following operations:
+
+- Obtain a connection to the current primary.
+- Step down the primary, and verify that another server has been elected as
+  a primary.
+- Dispatch an acknowledged insert operation over the connection obtained in
+  the first step.
+- Verify that the result of the insert operation is a NotMaster error.
+- Verify that the connection used for the insert operation remains open.
+
+Unacknowledged Write
+````````````````````
+
+This test requires server version 4.2 or higher.
+
+Perform the following operations:
+
+- Obtain a connection to the current primary.
+- Step down the primary, and verify that another server has been elected as
+  a primary.
+- Dispatch an unacknowledged insert operation over the connection obtained
+  in the first step.
+- Dispatch another unacknowledged insert operation over the connection obtained
+  in the first step.
+- If the second unacknowledged insert succeeded, dispatch another unacknowledged
+  insert over the same connection[*].
+- Verify that the last unacknowledged insert produced a network error due
+  to the connection being closed by the server.
+
+[*] Depending on buffering performed by the network stack, the first write
+operation on a connection which was closed on the remote end may succeed.
+In the acknowledged write scenario, the error is detected upon (trying to)
+read the response. In the unacknowledged write scenario, an additional write
+may be needed to detect that the connection is closed.
+
+Primary-Secondary-Primary Cycle
+```````````````````````````````
+
+This test requires server version 4.2 or higher.
+
+Perform the following operations:
+
+- Obtain a connection to the current primary.
+- Step down the primary, and verify that another server has been elected as
+  a primary.
+- Dispatch an acknowledged insert operation over the connection obtained in
+  the first step.
+- Verify that the result of the insert operation is a NotMaster error.
+- Verify that the connection used for the insert operation remains open.
+- Step up the server which was originally the primary.
+- Dispatch another acknowledged insert operation over the connection obtained
+  in the first step.
+- Verify that the insert succeeded and the data inserted is in the collection.
```

Comment:
Per the [SDAM spec](https://github.com/mongodb/specifications/blob/master/source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst#not-master-and-node-is-recovering), encountering a NotMaster error should update the server's state to Unknown. Is this test expecting the driver to bypass server selection and execute the insert against the server while it has an Unknown state?

If so, do I assume correctly that the server would maintain its Unknown state until the next monitoring interval?

Reply:
Good point, I believe I need to modify the procedure to stop sdam monitoring in this case, so that the driver thinks the no-longer-primary server is still the primary.